game/play_game.go

```diff
@@ -0,0 +1,188 @@
+package game
+
+import (
+	"fmt"
+	"sync"
+)
+
+func (g *Game) Play() error {
+	for !g.IsOver() {
+		var err error
+		switch g.round.CurrentStage {
+		case Deal:
+			err = g.dealPhase()
+			g.round.CurrentStage = BuildCrib
+		case BuildCrib:
+			err = g.buildCrib()
+			g.round.CurrentStage = Cut
+		case Cut:
+			err = g.cutPhase()
+			g.round.CurrentStage = Pegging
+		case Pegging:
+			err = g.peg()
+			g.round.CurrentStage = Counting
+		case Counting:
+			fmt.Printf("Counting players hands\n")
+			err = g.countHands()
+			g.round.CurrentStage = CribCounting
+		case CribCounting:
+			fmt.Printf("Counting the crib\n")
+			g.countCrib()
+			g.round.CurrentStage = Done
+		case Done:
+			err = g.NextRound()
+		}
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (g *Game) dealPhase() error {
+	// shuffle the deck at least once
+	g.Deck().Shuffle()
+
+	// init dealer
+	d := g.Dealer()
+	d.TakeDeck(g.Deck())
+
+	// shuffle
+	d.Shuffle()
+
+	// deal
+	ps := g.PlayersToDealTo()
+	err := deal(d, ps)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func deal(d Player, ps []Player) error {
+	for everyoneIsHappy := false; !everyoneIsHappy; {
```

Comment:
we should probably switch this logic to something like
```go
numCardsToDeal := 6 * 2
if len(ps) == 3 {
  numCardsToDeal = 5 * 3
} else if len(ps) == 4 {
  numCardsToDeal = 4 * 4
}
```
and then add a card to the crib for 3 player games